go/arrow/array/fixedsize_binary.go

```diff
@@ -0,0 +1,45 @@
+package array
+
+import "github.com/apache/arrow/go/arrow"
+
+// A type which represents an immutable sequence of fixed-length binary strings.
+type FixedSizeBinary struct {
```

Comment:
perhaps add the usual trick to test that `FixedSizeBinary` implements `array.Interface` ?